### PR TITLE
Updates Yams 3.0.0 -> 4.0.0

### PR DESCRIPTION
**Pull request by markotect**
* [x] I've started my branch from the `develop` branch (gitflow)
  * [x] And I've selected `develop` as the "base" branch for this Pull Request I'm about to create
* [x] I've added an entry in the `CHANGELOG.md` file to explain my changes and credit myself  
      _(or added `#trivial` to the PR title if I think it doesn't warrant a mention in the CHANGELOG)_
  * [x] Add the entry in the appropriate section (Bug Fix / New Feature / Breaking Change / Internal Change)
  * [x] Add a period and 2 spaces at the end of your short entry description
  * [x] Add links to your GitHub profile to credit yourself and to the related PR and issue after your description

---

Solves problem described in #772.

### CODE REVIEW

**Comment by SwiftGen-Eve** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should make sure that <code>Podfile.lock</code> and <code>Package.resolved</code> are changed in sync</td>
    </tr>
  </tbody>
</table>

Hey 👋 I'm Eve, the friendly bot watching over SwiftGen 🤖

Thanks a lot for your contribution!



---

Once you fix those tiny nitpickings above, we should be good to go! 🙌

ℹ️ _I will update this comment as you add new commits_

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,9 @@ _None_
 * Update the Swift version in `.swift-version` so that the right version is used when building manually (using `swiftenv`).  
   [@cfiken](https://github.com/cfiken)
   [#764](https://github.com/SwiftGen/SwiftGen/issues/764)
+* Update Yams version from `3.0.0` to `4.0.0` in order to use latest version of SwiftGen with other popular tools (such as SwiftLint) as Swift package dependencies.
```
Comment (djbe) [inline CHANGELOG.md]: This belongs under the "Internal changes", there's no real need to explain "why" we upgraded.

**Review by djbe — CHANGES_REQUESTED** [review]
Did you run pod update? Because I'm not seeing any project/source changes.

File: CHANGELOG.md (hunk 1 above)
Comment (markotect) [reply CHANGELOG.md]: Makes sense, moved it.

**Review by djbe — APPROVED** [review]

**Pull request merged**